### ISSUE TITLE
Bug: IE 11 throws an exception when HTML contains an HTML entity

### ISSUE THREAD

**Issue by joshcampbell191** [surveyjs/survey-pdf#37]
### Are you requesting a feature, reporting a bug or asking a question?
Reporting a bug.

### What is the current behavior?
If an html element contains an html entity (ex. `&rsquo;`), it fails to convert the [HTML to image](https://github.com/surveyjs/survey-pdf/blob/54864951bc129c0de694971f4cdd3e3eeeaa18c1/src/helper_survey.ts#L236) since IE doesn't support [ChildNode.remove](https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/remove). It got further when using [element-remove-polyfill](https://github.com/0x11-dev/element-remove-polyfill#readme), however I can't remember if it was able to complete successfully.

### What is the expected behavior?
It should convert the element without an exception.

### How would you reproduce the current behavior (if this is a bug)?
1. Print the PDF from the tested page in IE 11

#### Provide the test code and the tested page URL (if applicable)

Tested page URL: https://plnkr.co/edit/sB7at2i6UDDDTMvA

Test code

```
Survey
    .StylesManager
    .applyTheme("modern");

var json = {
    "pages": [
        {
            "questions": [
                {
                    "type": "html",
                    "html": "test&rsquo;s"
                }
            ]
        }
    ]
};

window.survey = new Survey.Model(json);

survey
    .onComplete
    .add(function (result) {
        document
            .querySelector('#surveyResult')
            .textContent = "Result JSON:\n" + JSON.stringify(result.data, null, 3);
    });

$("#surveyElement").Survey({model: survey});

function saveSurveyToPdf(filename, surveyModel, pdfWidth, pdfHeight) {
    var options = {
        fontSize: 14,
        margins: {
            left: 10,
            right: 10,
            top: 10,
            bot: 10
        },
        format: [pdfWidth, pdfHeight]
    };
    var surveyPDF = new SurveyPDF.SurveyPDF(json, options);
    surveyPDF.data = surveyModel.data;
    surveyPDF.save(filename);
}

document
    .getElementById("saveToPDFbtn")
    .onclick = function () {
        var pdfWidth = survey.pdfWidth || 210;
        var pdfHeight = survey.pdfHeight || 297;
        saveSurveyToPdf("surveyResult.pdf", survey, pdfWidth, pdfHeight);
    };
```

### Specify your

* browser: Internet Explorer 11
* browser version: OS Build 19042.870
* surveyjs platform (angular or react or jquery or knockout or vue): jQuery
* surveyjs version: 1.8.41

**Comment by gologames** [comment]
Hello,

We investigated the issue. Please, let me describe what we know now

First of all for proper work `jsPDF` library (which is used by SurveyPDF) you should load on page [jsPDF IE polyfill](https://unpkg.com/jspdf@2.3.1/dist/polyfills.umd.js). Due to some additional errors you might also use [polyfill for "remove" method](https://unpkg.com/element-remove-polyfill@1.1.0/index.js) (probably with additional check in case of `parentNode` is `null` or modify source code of `htmlToImage` method below), ["Array.from" method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#polyfill) and something else

Next we faced with `SecurityError` when call `canvas.toDataURL` in our [HTML to image source code](https://github.com/surveyjs/survey-pdf/blob/5f6bf6bf1352a99a4f5dac30074d0562b98583ff/src/helper_survey.ts#L274). We tried set `crossOrigin` property of `Image` instance to `"anonymous"`, but this doesn't help. Most likely because we put `svg` element image and [canvas became tainted](https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image#security_and_tainted_canvases)

To escape this we attempted to use [canvg library](https://www.npmjs.com/package/canvg), but it didn't work out for us. In exported PDF file we see nothing or black screen (`ignoreClear` option and other techniques don't help)

So as a result, unfortunately, at this moment we have no idea how process issue with exporting HTML to image in IE11. You can try to find a solution and make live sample for us (fork repository or reassign `SurveyHelper.htmlToImage` method for example). In this case we will investigate it and probably merge with main branch

In conclusion as a workaround you can use Edge browser (we checked this case works fine in Edge) or export [HTML via our standard render](https://surveyjs.io/Documentation/Pdf-Export?id=HtmlToPdf) with replacing `&rsquo;` to `'` before export according to [this convention](https://www.w3.org/wiki/Common_HTML_entities_used_for_typography). Here is [live sample of this approach](https://plnkr.co/edit/EWqNhbXD9JoJtpbn) (works with IE 11):

![image](https://user-images.githubusercontent.com/30930391/114024062-4b340880-987c-11eb-9f09-73086ae894b9.png)


Thanks, Alex
SurveyJS Team

**Comment by joshcampbell191** [comment]
Thanks for the detailed response and the workaround. We've gone ahead and replaced the instances of HTML entities to avoid the conversion to image within our survey for the time being. I'll keep an eye out for other potential solutions when time permits.

**Comment by gologames** [comment]
Hello,

Fine! Feel free to contact us in case of any further questions

Thanks, Alex
SurveyJS Team

**Comment by gologames** [comment]
Note for SurveyJS:

Issue can occurs due to we create new document instance (when convert to XML). New document instance has different CORS
